src/main/java/Storage.java

```diff
@@ -0,0 +1,117 @@
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Scanner;
+import java.util.ArrayList;
+import java.io.FileWriter;
+import java.io.File;
+
+/**
+ * Storage class that deals with loading tasks from a file
+ * and saving tasks into a file.
+ */
+public class Storage {
+
+    private File file;
+
+    /**
+     * Storage class constructor.
+     */
+    public Storage() {
+        String rootDir = new File(System.getProperty("user.dir")).getParentFile().getPath();
+        StringBuilder path = new StringBuilder();
+        path.append(rootDir);
+
+        File dir = new File(path + "/duke/data");
+
+        if (!dir.exists()) {
+            dir.mkdirs();
+        }
+
+        this.file = new File(dir + "/duke.txt");
+        System.out.println(rootDir);
+    }
+
+    /**
+     * Method which allows Storage class to read the file and load tasks
+     * that are stored in the file.
+     * @return ArrayList of tasks that stores all tasks written in the file.
+     */
+    public ArrayList<Task> readFile() {
+        ArrayList<Task> dukeList = new ArrayList<>();
+        try {
+            Scanner s = new Scanner(file);
+            while (s.hasNextLine()) {
+                String[] command = s.nextLine().split("/");
+
+                String taskType = command[0].trim();
+                boolean isDone;
+                if (command[1].trim().equals("1")) {
+                    isDone = true;
+                } else {
+                    isDone = false;
+                }
+
+                switch (taskType) {
+                case "deadline":
+                    String description = command[2].trim();
+                    String by = command[3].trim();
```

Comment:
Same as above, can use constants like `static final String DESCRIPTION_INDEX = 2;` to make the code more self-explanatory.